### PR TITLE
Money Over 9999 Display Issue #136

### PR DESCRIPTION
**Pull request by scirelli**
# Description
* Attempted to fix Issue #136
* Created a `drawStringLeftAligned` method to left align text.

I'm not sure if this was the correct way to do this. But from what I could tell the only options were to center or not center. The previous code was shifting the centered text off to the left. Since the text was drawn from the center of the string, when the money string got long enough it would draw over the money icon.

# Testing
* See issue #136 to reproduce the draw over.
* Edit your prevShip.ini, change the money value to something over 9999. It should no longer render over the icon, and remain left aligned.
* Change it to the max value 2147483647, it should still not draw over the icon.

# Notes
I ran the default Intellij's code formatter. I have my tabs set to 4 spaces. You don't have any documentation on code format. I see 2 space tabs, and I think I see 4 space ones as well. let me know which one you prefer. 

### Pre Pull Request Checklist:

- [x] Code has been scanned with [SonarLint](http://www.sonarlint.org/intellij/)
- [ ] Methods have been annotated with @ Nullable and @ Nonnull ([Read more](https://github.com/MovingBlocks/DestinationSol/wiki/@Nullable-&-@Nonnull-Quickstart))
- [x] There are no errors present in the project
- [x] Code has been formatted and indented
- [ ] Methods have appropriate Javadoc ([How to write Javadoc](http://www.oracle.com/technetwork/java/javase/documentation/index-137868.html))

### CODE REVIEW

**Comment by GooeyHub** [comment]
Can one of the admins please verify this patch?

File: main/src/org/destinationsol/CommonDrawer.java [hunk 1]
```diff
@@ -29,124 +29,139 @@
...
+    public void begin() {
+        mySpriteBatch.begin();
+    }
+
+    public void end() {
+        myTextureChecker.onEnd();
+        mySpriteBatch.end();
+    }
+
+    public void drawString(String s, float x, float y, float fontSize, Color col) {
```
Comment (scirelli) [inline main/src/org/destinationsol/CommonDrawer.java]: Removed the boolean `centered`.

File: main/src/org/destinationsol/CommonDrawer.java [hunk 2]
```diff
@@ -29,124 +29,139 @@
...
+        if (s == null) return;
+        myTextureChecker.onString(myFont.getRegion().getTexture());
+        myFont.setColor(col);
+        myFont.getData().setScale(fontSize / myOrigFontHeight);
+        myFont.draw(mySpriteBatch, s, x, y);
+    }
+
+    public void drawStringCentered(String s, float x, float y, float fontSize, Color col) {
```
Comment (scirelli) [inline main/src/org/destinationsol/CommonDrawer.java]: Created a separate method for centering text.

File: main/src/org/destinationsol/CommonDrawer.java [hunk 3]
```diff
@@ -29,124 +29,139 @@
...
+
+        // http://www.badlogicgames.com/wordpress/?p=3658
+        layout.reset();
+        layout.setText(myFont, s);
+        x -= layout.width / 2;
+        y -= layout.height / 2;
+        myFont.draw(mySpriteBatch, layout, x, y);
+    }
+
+    public void drawStringLeftAligned(String s, float x, float y, float fontSize, Color col) {
```
Comment (scirelli) [inline main/src/org/destinationsol/CommonDrawer.java]: Added a method to left align text. Just doesn't shift the text to the horizontal center of the layout.

**Comment by Cervator** [comment]
@GooeyHub: add to whitelist

Thanks again @scirelli - poking @Rulasmur for review since he has some internet again he should be putting to good use :D

We might still be doing a mass-formatting submit here in the near future, so don't worry too much about conventions just yet

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by vampcat** [comment]
Checks out great!

However, and this is more long term thinking than anything, drawString draws is without any centering whatsoever, drawStringLeftAligned draws is with only vertical centering, drawStringCenterAligned draws it with vertical and horizontal centering.. I think in the future, we can make a common
`drawString(..., boolean horizontalCentering, boolean verticalCentering, ...);`

**Comment by scirelli** [comment]
Yea, I wasn't sure what would be useful. The way I saw it being used I figured you'd always want it vertically centered. I can make a common drawString, with a TextAlignment enum.

**Comment by scirelli** [comment]
Closing this an opening a new PR #142

**Pull request closed without merging**